packages/cli/src/TagHelpers.ts

```diff
@@ -86,3 +88,67 @@ export async function removeRelations(workflowId: string, tablePrefix: string) {
 		.where('workflowId = :id', { id: workflowId })
 		.execute();
 }
+
+const createTag = async (transactionManager: EntityManager, name: string): Promise<TagEntity> => {
+	const tag = new TagEntity();
+	tag.name = name;
+	return transactionManager.save<TagEntity>(tag);
+};
+
+const findOrCreateTag = async (
+	transactionManager: EntityManager,
+	importTag: ITagToImport,
+	tagsEntities: TagEntity[],
+): Promise<TagEntity> => {
+	// Assume tag is identical if createdAt date is the same to preserve a changed tag name
+	const identicalMatch = tagsEntities.find(
+		(existingTag) =>
+			existingTag.id.toString() === importTag.id.toString() &&
+			existingTag.createdAt &&
+			importTag.createdAt &&
+			new Date(existingTag.createdAt) === new Date(importTag.createdAt),
+	);
+	if (identicalMatch) {
+		return identicalMatch;
+	}
+
+	// Find tag with identical name
+	const nameMatch = tagsEntities.find((existingTag) => existingTag.name === importTag.name);
+	if (nameMatch) {
+		return nameMatch;
+	}
+
+	// Create new Tag
+	const createdTag = await createTag(transactionManager, importTag.name);
+	return createdTag;
+};
+
+/**
+ * Set tag ids to use existing tags, creates a new tag if no matching tag could be found
+ *
+ * @param transactionManager
+ * @param workflow
+ * @param tagsEntities
+ * @returns
+ */
+export async function setTagsForImport(
+	transactionManager: EntityManager,
+	workflow: { tags: ITagToImport[] },
+	tags: TagEntity[],
+): Promise<void> {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+	const workflowTags = workflow.tags;
+	if (!workflowTags || !Array.isArray(workflowTags) || workflowTags.length === 0) {
+		return;
+	}
+	for (let i = 0; i < workflowTags.length; i++) {
+		if (workflowTags[i] && typeof workflowTags[i].name !== 'undefined') {
```

Comment:
1. The index is uneeded, so we can simplify for readability: `for (const tag of workflowTags)`. In fact, can parallelize with `Promise.all()`?
2. Simpler check: `workflowTags[i].name !== undefined`

Reply:
keeping index to replace imported tag in workflow